### PR TITLE
connected login form to API endpoint

### PR DESCRIPTION
**Pull request by Huayao-Zheng**
connected login form to API endpoint

### CODE REVIEW

**Review by qq1055394882 — APPROVED** [review]
reviewed and provided a feedback in discord.

File: e-commerce-ui/src/app/login/login.component.ts [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
     });
   }
-}
\ No newline at end of file
+
+  onLogin() {
+    this.http
+      .get(
+        `http://localhost:8080/api/v1/users/check-email?email=${this.emailForm.value.email}`
+      )
```
Comment (rzheng95) [inline e-commerce-ui/src/app/login/login.component.ts]: Move the login components into a `auth` folder. Create an authentication service (ex,. `auth.service.ts`) and put all authentication-related HTTP requests in the service file. You can generate the service files using ng command `ng g s auth/auth`

File: e-commerce-ui/src/app/login/login.component.ts [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
-    this.emailControl = new FormControl('example@example.com', [Validators.required, Validators.email]);
+  constructor(private http: HttpClient) {
+    this.emailControl = new FormControl('example@example.com', [
+      Validators.required,
+      Validators.email,
+    ]);
     this.emailForm = new FormGroup({
-      email: this.emailControl
+      email: this.emailControl,
     });
```
Comment (rzheng95) [inline e-commerce-ui/src/app/login/login.component.ts]: Inject FormBuilder and instantiate the `emailForm` with FormBuilder

**Review by rzheng95 — CHANGES_REQUESTED** [review]

File: e-commerce-ui/src/app/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(private http: HttpClient) {}
+
+  onLogin(email: string) {
```
Comment (rzheng95) [inline e-commerce-ui/src/app/auth/auth.service.ts]: Specify the return type.

File: e-commerce-ui/src/app/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    this.http
+      .get(`http://localhost:8080/api/v1/users/check-email?email=${email}`)
+      .subscribe(
+        (data) => {
+          console.log(data);
+        },
+        (error) => {}
+      );
```
Comment (rzheng95) [inline e-commerce-ui/src/app/auth/auth.service.ts]: Do not subscribe in service files. Only subscribe in components files.

File: e-commerce-ui/src/app/auth/login/login.component.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
+  <mat-form-field>
+    <mat-label>Email</mat-label>
+    <input matInput type="email" formControlName="email" />
+    <mat-error *ngIf="loginForm.get('email')?.hasError('required')"
+      >Email is required.</mat-error
+    >
+    <mat-error *ngIf="loginForm.get('email')?.hasError('email')"
+      >Invalid email format.</mat-error
+    >
```
Comment (rzheng95) [inline e-commerce-ui/src/app/auth/login/login.component.html]: Fix the formatting

**Review by rzheng95 — CHANGES_REQUESTED** [review]

**Review by rzheng95 — APPROVED** [review]

**Pull request merged**